include/spdlog/details/os-inl.h

```diff
@@ -398,28 +403,42 @@ SPDLOG_INLINE bool in_terminal(FILE *file) SPDLOG_NOEXCEPT
 #endif
 }
 
-#if defined(SPDLOG_WCHAR_TO_UTF8_SUPPORT) && defined(_WIN32)
-SPDLOG_INLINE void wbuf_to_utf8buf(const fmt::wmemory_buffer &wbuf, fmt::memory_buffer &target)
+#if (defined(SPDLOG_WCHAR_TO_UTF8_SUPPORT) || defined(SPDLOG_WCHAR_FILENAMES)) && defined(_WIN32)
+SPDLOG_INLINE void wstr_to_utf8buf(basic_string_view_t<wchar_t> wstr, fmt::memory_buffer &target)
 {
-    int wbuf_size = static_cast<int>(wbuf.size());
-    if (wbuf_size == 0)
+    if (wstr.size() > static_cast<size_t>(std::numeric_limits<int>::max()))
     {
+        throw spdlog::spdlog_ex("UTF-16 string is too big to be converted to UTF-8");
+    }
+
+    int wstr_size = static_cast<int>(wstr.size());
+    if (wstr_size == 0)
+    {
+        target.resize(0);
```

Comment:
seems like redundant resize

Reply:
I added so that if somebody passes a previously used memory_buffer, it'll be correctly emptied instead of staying filled.

Reply:
i meant the other resize , the one below

Reply:
anyway, it would be interesting to compare the new performance. could you run your benchmark again?